hawkbit-rest/hawkbit-mgmt-resource/src/test/java/org/eclipse/hawkbit/mgmt/rest/resource/MgmtDistributionSetResourceTest.java

```diff
@@ -273,6 +273,32 @@ public void assignMultipleTargetsToDistributionSet() throws Exception {
                 .as("Five targets in repository have DS assigned").hasSize(5);
     }
 
+    @Test
+    @Description("Ensures that targets can be assigned even if the specified controller IDs are in different case (e.g. 'TARGET1' instead of 'target1'.")
+    public void assignTargetsToDistributionSetIgnoreCase() throws Exception {
+        final DistributionSet createdDs = testdataFactory.createDistributionSet();
+
+        // prepare targets
+        final String[] knownTargetIds = new String[] { "64-da-a0-02-43-8b", "Trg1", "target2", "target4" };
+        final String[] knownTargetIdDifferentCase = new String[] { "64-DA-A0-02-43-8b", "TRG1", "TarGET2", "target4" };
+        for (final String targetId : knownTargetIds) {
+            testdataFactory.createTarget(targetId);
+        }
+        final JSONArray list = new JSONArray();
```

Comment:
You could also move it to `JsonBuilder` class, so it can be reused later

Reply:
For consistency reasons, I would prefer to stick to the current "manual" approach. That would also require JsonBuilder to offer a method like "targetsWithActionType" which returns the serialized JSON representation).